fish-rust/src/builtins/shared.rs

```diff
@@ -155,3 +157,54 @@ pub fn builtin_print_help(parser: &mut parser_t, streams: &io_streams_t, cmd: &w
         empty_wstring(),
     );
 }
+
+pub struct HelpOnlyCmdOpts {
+    pub print_help: bool,
```

Comment:
Given that we add `optind` here which feels a bit out of place, maybe it's better to replace the struct by a pair.

This ties in with another consideration:
You replace `parse_help_only_cmd_opts()` with `HelpOnlyCmdOpts::parse()`.
Whoever ports the next builtin that uses `parse_help_only_cmd_opts()` needs to figure out that it's called `HelpOnlyCmdOpts::parse()` now.

We already change type names (from `foo_bar_t` to `FooBar`) but that's less of an issue because those follow an obvious pattern.
So far we mostly preserve function names.

Actually this is likely not an issue here because it will be obvious how to port the earlier `struct help_only_cmd_opts_t`.
But it's something to keep in mind in general.

I'm guilty of the same thing, I replaced
`keyword_description()` with an implementation of `From<ParseKeyword> for &'static wstr`
`keyword_description()`  is still there but it's private and only used for FFI.
I've been thinking maybe I should make `keyword_description()` public again, and maybe add a TODO that it can be removed once we're done.

Summing up, I think your version is perfectly fine.
I think it's worth considering this alternative:

	let Ok((print_help, optind)) = parse_help_only_cmd_opts();